src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -45,7 +45,20 @@ internal static ZipkinSpan ToZipkinSpan(this Activity activity, ZipkinEndpoint l
                 Tags = PooledList<KeyValuePair<string, object>>.Create(),
             };
 
-            activity.EnumerateTags(ref tagState);
+            if (activity.Status == ActivityStatusCode.Ok || activity.Status == ActivityStatusCode.Error)
+            {
+                PooledList<KeyValuePair<string, object>>.Add(
+                    ref tagState.Tags,
+                    new KeyValuePair<string, object>(
+                        SpanAttributeConstants.StatusCodeKey,
+                        StatusHelper.GetTagValueForActivityStatusCode(activity.Status)));
+            }
+            else
+            {
+                // In the rare cases when both ActivityStatus and Status Tags were set,
+                // ActivityStatus takes precedence over status tags.
+                activity.EnumerateTags(ref tagState);
```

Comment:
This is still needed for other tags. There can be additional tags on activity.

Reply:
Addressed the issue by marking
``
tagState.StatusCode = StatusCode.Unset;
tagState.StatusDescription = string.Empty;
``
in the case when both activity status and status tag were set.
And updated the testcases to cover this scenario.
Thank you.